airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -37,11 +37,12 @@ class BigqueryStream(HttpStream, ABC):
     primary_key = "id"
```

Comment:
(This comment should be aligned with `BigqueryDatasets.parse_response`)

It seems like I can't run `discover` with our own dataset as it crashes with a `TypeError: 'NoneType' object is not iterable` because `records = response.json().get(self.name)` does not return an empty list but `None`. This seems to happen when there are no tables in the dataset. In my case, it happens with the request to `https://bigquery.googleapis.com/bigquery/v2/projects/dataline-integration-testing/datasets/EdgaoTestCaseSensitive/tables` where the dataset does not have any table.

For documentation purposes, here is the actual HTTP response:
```
{\n  \"kind\": \"bigquery#tableList\",\n  \"etag\": \"1B2M2Y8AsgTpgAmY7PhCfg==\",\n  \"totalItems\": 0\n}
```

Reply:
Yes, if there's no table in the dataset it would crash, do you think it's better if we change this behaviour?

Reply:
I updated `BigqueryTables` `parse_response` method to not fail when a table is empty, here's the code https://github.com/artefactory/airbyte/blob/233eac2e2d17fafe78111eb45226bcd293c66789/airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py#L127 
I also updated `BigqueryDatasets` `parse_response` method to raise an exception with the appropriate error message when no datasets are found here: https://github.com/artefactory/airbyte/blob/233eac2e2d17fafe78111eb45226bcd293c66789/airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py#L96